saga/saga_state.go

```diff
@@ -246,24 +246,20 @@ func (state *SagaState) addTaskData(taskId string, msgType SagaMessageType, data
 }
 
 /*
- * Applies the supplied message to the supplied sagaState.  Does not mutate supplied Saga State
- * Instead returns a new SagaState which has the update applied to it
+ * Applies the supplied message to the supplied sagaState.
+ * Mutates state only if msg is a valid transition.
```

Comment:
I believe the code currently mutates only after checking, but that's a hard thing to ensure we maintain, so I've removed the comment.